lib/Schedule/LongSteps.pm

```diff
@@ -455,4 +469,102 @@ sub find_process{
     return $self->storage()->find_process($pid);
 }
 
+sub load_process {
+    my ( $self, $pid, $context ) = @_;
+    $context ||= {};
+
+    my $loaded_process;
+    eval {
+        my $stored_process = $self->find_process($pid);
+        $loaded_process = $self->_load_process( $stored_process, $context );
+    };
+
+    if ($@) {
+        $log->critical("Failed to load the process: $@");
+    }
+
+    return $loaded_process;
+}
+
+sub revive {
+    my ( $self, $process_id, $revive_to ) = @_;
+
+    # find process
+    my $stored_process = eval { $self->find_process($process_id) };
+    if ($@) {
+        $log->critical("$process_id does not exist");
+        return;
+    }
+
+    # check to see if its allowed to be revived.
+    # has it stopped
+    if ( $stored_process->status() ne "terminated" ) {
+        $log->critical("$process_id does not have a status of 'terminated'");
+        return;
+    }
+
+    # load the process and check if process have the method to revive_to
+    # if revive $revive_to was not passed, used the function we failed on.
+    # and check that also, just in case we attempt to revive on a method
+    # that was previously removed.
+    $revive_to = $stored_process->what() unless $revive_to;
+    my $loaded_process = eval { $self->_load_process($stored_process) };
+
+    if ($@) {
+        $log->critical("Failed to load the process: $@");
+        return;
+    }
+
+    unless ($loaded_process->can($revive_to)) {
+        $log->critical("Unable revive $process_id to $revive_to");
+        return;
+    }
+
+    my $revive_method = "revive_$revive_to";
```

Comment:
don't need that.

What you can do is on the process base class:

An instance method that returns all the 'revive_*' methods.  (overridable in sub processes).